tasks/git-hooks/protected-branches.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/python3
+
+import re
+import subprocess
+import sys
+
+
+def main():
+    try:
+        local_branch = subprocess.check_output('git rev-parse --abbrev-ref HEAD', shell=True).decode('utf-8').strip()
+        print(local_branch)
+        if local_branch == 'main':
+            print("You're about to commit on main, are you sure this is what you want?")
+            sys.exit(1)
+        if re.fullmatch(r'^[0-9]+\.[0-9]+\.x$', local_branch):
+            print("You're about to commit on a release branch, are you sure this is what you want?")
+            sys.exit(1)
+    except OSError:
+        sys.exit(1)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Maybe we should print this error to avoid having a silence error ?

```suggestion
    except OSError as e:
        print(e)
        sys.exit(1)
```

Reply:
Yeah that's definitely better, done!